lib/app/register/register_view.dart

```diff
@@ -0,0 +1,175 @@
+                        ),
+                        GestureDetector(
+                          onTap: () => _setAgreedToTOS(!_agreedToTOS),
+                          child: Text(
+                            'I agree to the Terms of Services and Privacy Policy',
+                            style: TextStyle(
+                              color: Colors.white,
+                              fontSize: 12.0,
+                            ),
+                          ),
+                        ),
+                      ],
+                    ),
+                  ),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: <Widget>[
+                      GestureDetector(
+                        onTap: () {
+                          // TODO: Open the TOS for view
+                        },
+                        child: Text(
+                          'Terms of Service',
+                          style: TextStyle(
+                            fontSize: 16.0,
+                            color: Colors.redAccent,
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                  SizedBox(height: 60.0),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: <Widget>[
+                      GestureDetector(
+                        onTap: () {
+                          if (_agreedToTOS &&
+                              _formKey.currentState.validate()) {
+                            print('Registration send successful');
+                            Navigator.of(context).pop();
+                            _controller.register(context);
+                          } else {
+                            print('Registration send unsuccessful');
+                          }
+                        },
+                        child: Container(
+                          width: 320.0,
+                          height: 50.0,
+                          alignment: FractionalOffset.center,
+                          decoration: new BoxDecoration(
+                              color: Color.fromRGBO(230, 38, 39, 1.0),
+                              borderRadius: new BorderRadius.circular(25.0)),
+                          child: new Text("Register",
+                              style: new TextStyle(
+                                  color: Colors.white,
+                                  fontSize: 20.0,
+                                  fontWeight: FontWeight.w300,
+                                  letterSpacing: 0.4)),
+                        ),
+                      ),
+                    ],
+                  )
+                ],
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _setAgreedToTOS(bool value) {
+    setState(() {
+      _agreedToTOS = value;
+    });
```

Comment:
These should go in the `RegisterController`

Reply:
Not sure how to get setState in the controller.

Reply:
Use the `callHandler()` that is in the `View`
Pass it the method of the controller like the other examples

Reply:
Every `View` has this method
```
  void callHandler(Function fn) {
    setState(() {
      fn();
    });
  }
```
which allows you to call handlers in the `Controller` and set the state at the same time
@briceallard

Reply:
Fixed